File: data/data.code.lua.client/MsgTestPhp.lua
module("MsgTestPhp", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = MsgTestPhp})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.u16 = 0
end


function encode(self)
	local pack = packet.new()
	pack:writeU16(self.u16)
	return pack:readBytes()
end

function decode(self, pack)
	self.u16 = pack:readU16()
	return self
end

function getBytes(self)
	return self:encode()
end

return MsgTestPhp

File: data/golang/msg_test_php.go
package proto

import (
	"packet"
)

type MsgTestPhp struct {
	U16                      uint16
}

func (this *MsgTestPhp) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint16(this.U16)

	return pack.Encode(uint16(0))
}

func (this *MsgTestPhp) EncodeMsg() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint16(this.U16)

	return pack.ReadBytes()
}

func MsgTestPhpDecode(pack *packet.Packet) *MsgTestPhp {
	msgTestPhp := &MsgTestPhp{}

	msgTestPhp.U16 = pack.ReadUint16()
	return msgTestPhp
}
